src/Types/NullableType.php

```diff
@@ -12,8 +12,20 @@
     /** @var Type */
     private $type;
 
-    public function __construct(Nullable $type)
```

Comment:
The problem is that we allow injection of implementations of `Nullable` here, but `Nullable` does not extend `Type`:

https://github.com/spatie/typed/blob/2a90dc0d354024867df4028c0a197cf4f7c81945/src/Nullable.php#L9-L12

However, in `NullableType::getType()` we have a return type declaration for `Type`:

https://github.com/spatie/typed/blob/2a90dc0d354024867df4028c0a197cf4f7c81945/src/Types/NullableType.php#L29-L32